rllib/connectors/tests/test_agent.py

```diff
@@ -435,6 +435,44 @@ def test_connector_pipline_with_view_requirement(self):
             t += 1
         print(total_rewards)
 
+    def test_vr_connector_only_keeps_useful_timesteps(self):
+        """Tests that the connector respects the flags within view_requirements (i.e.
+        used_for_training, used_for_compute_actions).
+
+        the returned data is the input dict itself, which the policy collector in
+        env_runner will use to construct the episode, and a SampleBatch that can be
+        used to run corresponding policy.
+        """
+        view_rqs = {
+            "obs": ViewRequirement(
+                None, used_for_training=True, used_for_compute_actions=True
+            ),
+        }
+
+        config = PPOConfig().to_dict()
+        ctx = ConnectorContext(
+            view_requirements=view_rqs,
+            config=config,
+            is_policy_recurrent=False,
+        )
+
+        c = ViewRequirementAgentConnector(ctx)
+        c.in_training()
+
+        for i in range(5):
+            obs_arr = np.array([0, 1, 2, 3]) + i
+            agent_data = {SampleBatch.NEXT_OBS: obs_arr}
+            data = AgentConnectorDataType(0, 1, agent_data)
+
+            # Feed ViewRequirementAgentConnector 5 samples.
+            c([data])
+
+        obs_data = c.agent_collectors[0][1].buffers["obs"][0]
```

Comment:
Makes sense. Curious, what was the behavior of obs_data before you make this `is_training=False`?

Reply:
goes up to however long the episode is.
and then during each timestep, we convert the whole episode into numpy ...

Reply:
I see.